### ISSUE TITLE
Odnoklassinki - Unspecified error. #6

### ISSUE THREAD

**Issue by artemrizhov** [miled/wordpress-social-login#30]
Can't login via Odnoklassniki because of this error. I followed the link from the debug output:

http://api.odnoklassniki.ru/fb.do?application_key=&method=users.getCurrentUser&sig=07aa4538101f8bad57a0a121b60612d2&access_token=57ipa.51-1c8y0282a6g31136b3k1bsrva8

And got this message:
<error_code>101</error_code>
<error_msg>PARAM_API_KEY : Application not exist</error_msg>

So the problem probably is the empty application_key parameter value.

Is there any workaround? Thanks.

**Comment by olegabr** [comment]
The problem is the emptiness of the `$this->config` array. The workaround is to replace all occurrences of the `$this->config['keys']['key']` with the public key of your ok.ru application in the getUserProfile function in the file https://github.com/miled/wordpress-social-login/blob/master/hybridauth/Hybrid/Providers/Odnoklassniki.php#L147. (You can find it on this path in your wordpress installation: `wp-content/plugins/wordpress-social-login/hybridauth/Hybrid/Providers/Odnoklassniki.php`) e.g.
```php
                $key_temp = 'CAASBCFFABABBBABL';

                $sig = md5('application_key=' . $key_temp . 'method=users.getCurrentUser' . md5($this->api->access_token . $this->api->client_secret));

                $response = $this->api->api( '?application_key=' . $key_temp . '&method=users.getCurrentUser&sig=' .$sig);
```

Note the `$key_temp` usage here.

Dear plugin developers, can you please elaborate on this issue? It seems that config usage has changed and the Odnoklassniki provider was not updated to use the new scheme?

**Comment by AlexeyKot** [comment]
Thanks, olegabr, you saved me a day!
Dear Developers, it's 2017, but the problem is still here - ok.ru service doesn't work out of the box.

**Comment by nineseo** [comment]
2019. Odnoklassniki (ok.ru) still does not work. Can anyone suggest how to fix version 3.0.1?
